src/Factory/Psr17Factory.php

```diff
@@ -20,6 +20,11 @@ public function createRequest(string $method, $uri): RequestInterface
 
     public function createResponse(int $code = 200, string $reasonPhrase = ''): ResponseInterface
     {
+        if (2 > \func_num_args()) {
```

Comment:
This is interesting, never thought of doing it this way! Basically a trick to emulate the method signature `string $reasonPhrase = null` without breaking the interface implementation?

Reply:
Yeah, basically. 

I check to see how many arguments that were actually used.